File: demo/java/view/src/main/java/pers/shawxingkwok/kdatastore/demo/viewjava/MainActivity.java
package pers.shawxingkwok.kdatastore.demo.viewjava;

import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.app.AppCompatDelegate;
import pers.shawxingkwok.kdatastore.demo.settings.Settings;
import pers.shawxingkwok.kdatastore.demo.viewjava.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    @SuppressWarnings("FieldCanBeLocal")
    private ActivityMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        // link `isDarkMode` with `theme` via `liveData`
        Settings.isDarkMode().getLiveData().observe(this, (isDark) -> {
            int mode;
            if (isDark == null)
                mode = AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM;
            else if (isDark)
                mode = AppCompatDelegate.MODE_NIGHT_YES;
            else
                mode = (AppCompatDelegate.MODE_NIGHT_NO);

            AppCompatDelegate.setDefaultNightMode(mode);
            getDelegate().applyDayNight();
        });

        // set the initial checked radioButton with `isDarkMode`
        if (Settings.isDarkMode().getValue() == null)
            binding.rbFollowSystem.setChecked(true);
        else if (Settings.isDarkMode().getValue())
            binding.rbDark.setChecked(true);
        else
            binding.rbLight.setChecked(true);

        // update `isDarkMode` in Listener
        binding.rgTheme.setOnCheckedChangeListener((group, checkedId) -> {
            if (checkedId == R.id.rb_followSystem) {
                Settings.isDarkMode().setValue(null);
            } else if (checkedId == R.id.rb_dark) {
                Settings.isDarkMode().setValue(true);
            } else if (checkedId == R.id.rb_light) {
                Settings.isDarkMode().setValue(false);
            } else {
                throw new IllegalStateException("Unexpected value: " + checkedId);
            }
        });
    }
}

File: android/viewkt/src/main/java/pers/shawxingkwok/kdatastore/demo/viewkt/MainActivity.kt
package pers.shawxingkwok.kdatastore.demo.viewkt

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.app.AppCompatDelegate
import androidx.lifecycle.lifecycleScope
import kotlinx.coroutines.flow.launchIn
import kotlinx.coroutines.flow.onEach
import pers.shawxingkwok.kdatastore.demo.viewkt.databinding.ActivityMainBinding
import pers.shawxingkwok.settings.Settings

class MainActivity : AppCompatActivity() {

    private val binding by lazy { ActivityMainBinding.inflate(layoutInflater) }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(binding.root)

        // link `isDarkMode` to `theme` via flow
        Settings.isDarkMode.onEach {
            val mode = when(it){
                null -> AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM
                true -> AppCompatDelegate.MODE_NIGHT_YES
                false -> AppCompatDelegate.MODE_NIGHT_NO
            }
            AppCompatDelegate.setDefaultNightMode(mode)
            delegate.applyDayNight()
        }
        .launchIn(lifecycleScope)

        /* or via liveData
        Settings.isDarkMode.liveData.observe(this){
            val mode = when(it){
                null -> AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM
                true -> AppCompatDelegate.MODE_NIGHT_YES
                false -> AppCompatDelegate.MODE_NIGHT_NO
            }
            AppCompatDelegate.setDefaultNightMode(mode)
            delegate.applyDayNight()
        }
        */

        // set the initial checked radioButton
        when (Settings.isDarkMode.value) {
            null -> binding.rbFollowSystem
            true -> binding.rbDark
            false -> binding.rbLight
        }
        .isChecked = true

        // update `isDarkMode` in Listener
        binding.rgTheme.setOnCheckedChangeListener { _, id ->
            Settings.isDarkMode.value =
                when(id){
                    R.id.rb_followSystem -> null
                    R.id.rb_dark -> true
                    R.id.rb_light -> false
                    else -> throw Error()
                }
        }
    }
}